### PR TITLE
Fix ChargingBench

### LINKED ISSUE #4183: ChargingBench can recharge muliple stackable items.
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

To test this out I created a simple addon with a stackable item as the tool that needs to be recharged, I put a stack into the ChargingBench and it recharged them.

### 📑 Reproduction Steps

1. Get yourself a stack of stackable rechargable items
2. Throw them into the charging bench
3. A stack worth of these items will recharge, effectively making charging bench +3200% more efficient

### 💡 Expected Behavior

1. Don't recharge the ItemStack

### 📷 Screenshots / Videos

https://github.com/Slimefun/Slimefun4/assets/61906290/10dfdf62-072e-43a4-b7b9-3ebbb829916a

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.20.x

### ⭐ Slimefun version

Dev - 1145

### 🧭 Other plugins

Any addon, fixing this may allow addon creators more freedom over their Rechargeable items

### PR DESCRIPTION
**Pull request by Intybyte**
## Description

Making ChargingBench rechargeable only ItemStacks with 1 items

## Proposed changes

By adding this feature addon makers can make heir own Rechargeable any item they want and not be limited to tools

## Related Issues (if applicable)

Resolves  #4183

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ChargingBench.java [hunk 1]
```diff
@@ -44,6 +44,12 @@ protected void tick(Block b) {
         for (int slot : getInputSlots()) {
             ItemStack item = inv.getItemInSlot(slot);
 
+            if (item == null)
+                continue;
+
+            if (item.getAmount() != 1)
+                continue;
+
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ChargingBench.java:52]:
```suggestion
            if (item == null) {
                continue;
            }

            if (item.getAmount() != 1) {
                continue;
            }

```

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Review by J3fftw1 — APPROVED** [review]
LGTM

**Comment by JustAHuman-xD** [comment]
I did not want to dismiss the review aaaa

**Review by JustAHuman-xD — APPROVED** [review]
LGTM

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Pull request merged**